src/app/ai-model-card/ai-model-card-new/ai-model-card-new.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+
+import { AiModelCard } from '../ai-model-card';
+import { AiModelCardService } from '../ai-model-card.service';
+import { frameworks, operation_types, architectures, licenses } from 'src/app/ai-model-data';
+
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+import { MessageService } from 'primeng/api';
+
+@Component({
+  selector: 'app-ai-model-card-new',
+  templateUrl: './ai-model-card-new.component.html',
+  styleUrls: ['./ai-model-card-new.component.css']
+})
+export class AiModelCardNewComponent implements OnInit {
+  form: AiModelCard = new AiModelCard();
+
+  list_framework = frameworks;
+  list_operation_type = operation_types;
+  list_architecture = architectures;
+  list_license = licenses;
+
+  train_data: [key: string, val: string] = [null, null];
+  train_params: [key: string, val: string] = [null, null];
+
+  constructor(public config: DynamicDialogConfig,
+    private ref: DynamicDialogRef,
+    private card_service: AiModelCardService,
+    private message_service: MessageService) { }
+
+  ngOnInit() { this.form.trainingData = {}; this.form.trainingParameters = {}; }
+
+  cancel() { this.ref.close(); }
+
+  // add then reset training form data or parameters
+  add_training(type: string, key: string, val: string) {
+    if (type == "data") {
```

Comment:
Variable and method names should be in camel case if they are not constants (should be uppercase) or test method names (see https://google.github.io/styleguide/tsguide.html#naming)